src/components/Imip.vue

```diff
@@ -414,44 +414,60 @@ export default {
 				return
 			}
 
-			attendee.participationStatus = status
-			if (this.comment) {
-				attendee.setParameter(new Parameter('X-RESPONSE-COMMENT', this.comment))
-				vEvent.addProperty(new Property('COMMENT', this.comment))
+			this.loading = true
+
+			if (!this.isExistingEvent) {
+				try {
+					await calendar.createVObject(vCalendar.toICS())
+					await this.fetchExistingEvent(vEvent.uid, true)
+				} catch (error) {
+					showError(this.t('mail', 'Failed to save your participation status'))
+					logger.error('Failed to save event to calendar', {
+						error,
+						attendee,
+						calendar,
+						vEvent,
+						vCalendar,
+						existingEvent: this.existingEvent,
+					})
+				}
 			}
-			// TODO: implement an input for guests and save it to the attendee via X-NUM-GUESTS
 
-			this.loading = true
-			try {
-				if (this.isExistingEvent) {
+			if (this.isExistingEvent) {
+				attendee.participationStatus = status
+				if (this.comment) {
+					attendee.setParameter(new Parameter('X-RESPONSE-COMMENT', this.comment))
+					vEvent.addProperty(new Property('COMMENT', this.comment))
+				}
+
+				// TODO: implement an input for guests and save it to the attendee via X-NUM-GUESTS
+
+				try {
 					// TODO: don't show buttons if calendar is not writable
 					this.existingEvent.data = vCalendar.toICS()
 					await this.existingEvent.update()
-				} else {
-					await calendar.createVObject(vCalendar.toICS())
+					this.showMoreOptions = false
+				} catch (error) {
+					showError(this.t('mail', 'Failed to save your participation status'))
+					logger.error('Failed to save event to calendar', {
+						error,
+						attendee,
+						calendar,
+						vEvent,
+						vCalendar,
+						existingEvent: this.existingEvent,
+					})
 				}
-				this.showMoreOptions = false
-			} catch (error) {
-				showError(this.t('mail', 'Failed to save your participation status'))
-				logger.error('Failed to save event to calendar', {
-					error,
-					attendee,
-					calendar,
-					vEvent,
-					vCalendar,
-					existingEvent: this.existingEvent,
-				})
-			} finally {
-				this.loading = false
```

Comment:
I liked unsetting loading in finally because this was it is ensured that the reset happens even on error

Reply:
I still reset, just a few lines later. We have two try-catch blocks now, and both are wrapped in an if statement. First the loading state is enabled, then one or both try-catch statements run, and last the loading state is disabled. 

The current loading state does not work because ButtonVue (an alias for NcButton) has no property called "loading". I've changed the code to disable the button and added the loading icon to the right.